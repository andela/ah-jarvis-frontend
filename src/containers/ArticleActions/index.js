/* eslint-disable */
import swal from 'sweetalert';
import M from 'materialize-css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';

import chevrondown from '../../assets/icons/chevron-down.svg';
import { deleteArticle } from '../Home/actions';

class ArticleActions extends Component {
  componentDidMount() {
    const el = document.querySelectorAll(`.${this.props.slug}`);
    M.Dropdown.init(el);
  }

  handleDelete = () => {
    const { slug, removeArticle } = this.props;

    swal({
      title: 'Are you sure?',
      text: 'Once deleted, you will not be able to recover this story.',
      icon: 'warning',
      buttons: true,
      dangerMode: true,
    }).then(willDelete => {
      if (willDelete) {
        removeArticle(slug);
        swal('Poof! Your story was successfully deleted!', {
          icon: 'success',
        });
      } else {
        swal('Your sweet story is safe!');
      }
    });
  };

  render() {
    const { slug } = this.props;

    return (
      <React.Fragment>
        <p className={`article-actions right ${slug} dropdown-trigger`} data-target={slug}>
          <img src={chevrondown} alt="" />
        </p>

        <ul id={slug} className="dropdown-content drop--small">
          <li>
            <Link to={`/article/${slug}`}>View</Link>
          </li>
          <li className="divider" tabIndex="-1" />
          <li>
            <Link to={`/article/${slug}/edit`}>Edit</Link>
          </li>
          <li className="divider" tabIndex="-1" />
          <li>
            <a onClick={this.handleDelete}>Delete</a>
          </li>
        </ul>
      </React.Fragment>
    );
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      removeArticle: deleteArticle,
    },
    dispatch,
  );

export default connect(
  null,
  mapDispatchToProps,
)(ArticleActions);

ArticleActions.propTypes = {
  slug: PropTypes.string.isRequired,
  removeArticle: PropTypes.func.isRequired,
};
