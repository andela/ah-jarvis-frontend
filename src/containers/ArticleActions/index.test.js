import { shallow } from 'enzyme';
import React from 'react';
import ArticleActions from '.';
import configStore from '../../store';

describe('<ArticleActions />', () => {
  it('renders  <ArticleActions /> components', () => {
    const snap = shallow(<ArticleActions store={configStore()} slug="" />);
    expect(snap).toMatchSnapshot();
  });
});
