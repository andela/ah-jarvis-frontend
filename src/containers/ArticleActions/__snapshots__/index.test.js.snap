// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleActions /> renders  <ArticleActions /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ArticleActions)
    slug=""
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "removeArticle": [Function],
      "slug": "",
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "removeArticle": [Function],
        "slug": "",
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
